packages/mobile/src/components/premium-track-purchase-drawer/StripePurchaseConfirmationButton.tsx

```diff
@@ -1,74 +1,64 @@
 import { useCallback } from 'react'
 
-import { accountSelectors } from '@audius/common'
+import type { ID } from '@audius/common'
+import {
+  purchaseContentActions,
+  ContentType,
+  purchaseContentSelectors,
+  PurchaseContentStage
+} from '@audius/common'
 import { useSelector, useDispatch } from 'react-redux'
 
 import { Button } from 'app/components/core'
-import { useNavigation } from 'app/hooks/useNavigation'
-import { createStripeSession, getUSDCUserBank } from 'app/services/buyCrypto'
-import { setVisibility } from 'app/store/drawers/slice'
 import { useThemeColors } from 'app/utils/theme'
 
-const { getAccountERCWallet } = accountSelectors
+import LoadingSpinner from '../loading-spinner/LoadingSpinner'
+
+const { getPurchaseContentFlowStage } = purchaseContentSelectors
+const { startPurchaseContentFlow } = purchaseContentActions
 
 const messages = {
-  buy: (price: string) => `Buy $${price}`
+  buy: (price: string) => `Buy $${price}`,
+  purchasing: 'Purchasing'
 }
 
 type StripePurchaseConfirmationButtonProps = {
+  trackId: ID
   price: string
 }
 
 export const StripePurchaseConfirmationButton = ({
+  trackId,
   price
 }: StripePurchaseConfirmationButtonProps) => {
   const dispatch = useDispatch()
-  const navigation = useNavigation()
   const { specialLightGreen1 } = useThemeColors()
-  const ethWallet = useSelector(getAccountERCWallet)
+  const stage = useSelector(getPurchaseContentFlowStage)
+  const isLoading =
```

Comment:
There's a [util in common ](https://github.com/AudiusProject/audius-client/blob/41cc4af1a1d329186afe8ff2b8452571b1745cf8/packages/common/src/store/purchase-content/utils.ts#L3)now that you can use for this.